### ISSUE TITLE
Feature: normalize errors from the Bitcoin backend on transaction broadcast

### ISSUE THREAD

**Issue by darosior** [ElementsProject/lightning#3970]
This would allow us to take action such as safely forgetting a channel creation attempt (see https://github.com/ElementsProject/lightning/pull/3910).

**Comment by ZmnSCPxj** [comment]
Does this mean potentially standardizing some error codes that `sendrawtransction` plugin command should report back?

**Comment by darosior** [comment]
Yep, i think this should not use the jsonrpc errors but a new field in the result to keep forward compatibility
-------- Original Message --------

**Comment by ZmnSCPxj** [comment]
Seems okay.  Any concrete thoughts on what results to return, other than e.g.  "fee below min relay"?  How about consensus failures?  And how does `sendpsbt` change to propagate this?

**Comment by darosior** [comment]
I would like the error codes to englobe different errors (for example, "below min mempool fee", "below min relay" (won't happen), "mempool full") so that we know that it's safe to cancel a channel after receiving a given class of errors. No concrete thought yet though.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by ZmnSCPxj** [comment]
Basically we want to have two classes of failures:

* The tx was never, ever broadcasted, because it never entered our own mempool in the first place.
* The tx *might* be broadcasted.  This includes the edge case where the connection between this node and the `bitcoind` is interrupted, because it is possible the `bitcoind` adds it to the mempool but the success response was not received by our machine.

Would that be a fair restatement?

**Comment by darosior** [comment]
It is a fair restatement, however i would like to keep it more generalistic than the "cancel a funding attempt" usecase. We may explicit for each class of error if the transaction might have been broadcast or not.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by ZmnSCPxj** [comment]
It seems to me that the error "this transaction was never broadcasted" could apply to far more than just `fundchannel`.  For example, `withdraw` as well would probably benefit from cancelling the transactions rather than retrying.

**Comment by ZmnSCPxj** [comment]
On the other hand, people will play around with PSBTs, and we might want to assist developers building on top of PSBTs.  For example, an incrrrect creation of PSBTs might result in a transaction that is rejected at consensus, so it might be useful to differentiate between "this tx is never broadcasted because you suck at making consensus-respecting transactions" vs "this tx is never broaddcasted because it will not fit the mempool".

**Comment by sangbida** [comment]
This issue is being closed as part of issue cleanup.

**Reason**: Very old issue (5.5 years old) no recent activity

If you believe this issue should remain open, please comment and we can reopen it.
